client/my-sites/checkout/checkout/web-payment-box.jsx

```diff
@@ -350,6 +352,47 @@ export class WebPaymentBox extends React.Component {
 		notices.error( errorMessage );
 	};
 
+	/**
+	 * @param {string} fieldName The name of the country select field.
+	 * @param {string} selectedCountryCode The selected country.
+	 */
+	updateSelectedCountry = ( fieldName, selectedCountryCode ) => {
+		setTaxCountryCode( selectedCountryCode );
+
+		// Apple Pay needs the country where the payment will be processed
+		// (https://developer.apple.com/documentation/apple_pay_on_the_web/applepayrequest/2951833-countrycode),
+		// so call the Paygate configuration API endpoint to get that. This
+		// information is only needed when the user clicks the button to buy
+		// something and opens the Apple Pay payment sheet, so it is somewhat
+		// wasteful to call it here (every time a new country is selected);
+		// however, it is necessary because browsers will only open a payment
+		// sheet in direct response to user input (see
+		// https://developer.mozilla.org/en-US/docs/Web/API/PaymentRequest/show),
+		// so we can't delay the opening of the payment sheet on waiting for
+		// the response to this API call.
+		wpcom.undocumented().paygateConfiguration(
+			{
+				request_type: 'new_purchase',
+				country: selectedCountryCode,
+				// The endpoint also accepts an optional credit card brand in
+				// cases where the processor country might depend on the card
+				// brand used, but unfortunately we can't send that because we
+				// don't know it yet (it's a chicken and egg problem; the user
+				// only selects a credit card once the Apple Pay payment sheet
+				// is open, but we need to send Apple Pay the processor country
+				// in order to open the payment sheet). Fortunately, in most
+				// cases this won't have a practical effect.
+			},
+			function( configError, configuration ) {
+				let processorCountry = 'US';
+				if ( ! configError && configuration.processor === 'stripe_ie' ) {
+					processorCountry = 'IE';
```

Comment:
A possible improvement would be to change the Paygate configuration API endpoint server-side so this information is returned as part of the response, rather than calculated here.